modules/apps/27-interchain-accounts/types/packet.go

```diff
@@ -16,7 +16,7 @@ func (iapd InterchainAccountPacketData) ValidateBasic() error {
 		return sdkerrors.Wrap(ErrInvalidOutgoingData, "packet data type cannot be unspecified")
 	}
 
-	if iapd.Data == nil {
+	if len(iapd.Data) == 0 {
```

Comment:
Will this cause nil pointer exception?

Reply:
This was done explicitly: https://github.com/cosmos/ibc-go/pull/1882 . This won't cause nil pointer exception, in go `len()` is safe to check on nil slices => it just returns 0.

Reply:
sounds good